slp2mp4/slp-to-mp4.py

```diff
@@ -40,10 +41,117 @@
 else:
     THIS_CONFIG = os.path.join(SCRIPT_DIR, 'config.json')
 THIS_USER_DIR = os.path.join(SCRIPT_DIR, 'User')
+OUT_DIR = os.path.join(SCRIPT_DIR, 'out')
+
+
+combined_files = []
+
 
 def is_game_too_short(num_frames, remove_short):
     return num_frames < MIN_GAME_LENGTH and remove_short
 
+
+# Evaluate whether file should be run. The open in dolphin and combine video and audio with ffmpeg.
+def record_file_slp(slp_file, outfile):
+    conf = Config()
+
+    # Parse file with py-slippi to determine number of frames
+    slippi_game = Game(slp_file)
+    num_frames = slippi_game.metadata.duration + DURATION_BUFFER
+
+    if is_game_too_short(slippi_game.metadata.duration, conf.remove_short):
+        print("Warning: Game is less than 30 seconds and won't be recorded. Override in config.")
+        return
+
+    # Dump frames
+    with DolphinRunner(conf, THIS_USER_DIR, SCRIPT_DIR, JOB_ID) as dolphin_runner:
+        video_file, audio_file = dolphin_runner.run(slp_file, num_frames)
+
+        # Encode
+        ffmpeg_runner = FfmpegRunner(conf.ffmpeg)
+        ffmpeg_runner.run(video_file, audio_file, outfile)
+
+        print('Created {}'.format(outfile))
+
+
+# In the out folder, run through each subdirectory and count the number of mp4 files. Add these files to
+# concat_file.txt. ffmpeg uses this to combine the mp4 files.
+def combine(conf):
+    for subdir, dirs, files in os.walk(OUT_DIR, topdown=False):
```

Comment:
topdown shouldn't matter I realised; record_folder_slp will re-create any deleted directories, so the order of traversal won't matter

Reply:
Yep